internal/openshift/openshift.go

```diff
@@ -519,3 +525,109 @@ func (oe *openshiftClient) GetDeployment(ctx context.Context, name string, names
 	}
 	return &deployment, nil
 }
+
+// GetDeploymentPods can return an ErrNotFound
+func (oe *openshiftClient) GetDeploymentPods(ctx context.Context, name string, namespace string) ([]corev1.Pod, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+
+	deployment, err := oe.GetDeployment(ctx, name, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	selectorLabels := deployment.Spec.Selector.MatchLabels
+	if len(selectorLabels) == 0 {
+		logger.V(log.TRC).Info("deployment has no selector labels defined, returning empty pod list", "namespace", namespace, "name", name)
+		return []corev1.Pod{}, nil
+	}
+
+	labelSelector := crclient.MatchingLabels{}
+	maps.Copy(labelSelector, selectorLabels)
+
+	podList := corev1.PodList{}
+	err = oe.Client.List(ctx, &podList, labelSelector, crclient.InNamespace(namespace))
+	if err != nil {
+		return nil, fmt.Errorf("could not list pods matching label selector: %v", err)
+	}
+
+	return podList.Items, nil
+}
+
+// GetPod can return an ErrNotFound
+func (oe *openshiftClient) GetPod(ctx context.Context, name string, namespace string) (*corev1.Pod, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+
+	logger.V(log.TRC).Info("fetching pod", "namespace", namespace, "name", name)
+	pod := corev1.Pod{}
+	err := oe.Client.Get(ctx, crclient.ObjectKey{
+		Name:      name,
+		Namespace: namespace,
+	}, &pod)
+	if apierrors.IsNotFound(err) {
+		return nil, fmt.Errorf("could not retrieve pod: %s/%s: %w: %v", namespace, name, ErrNotFound, err)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve pod: %s/%s: %v", namespace, name, err)
+	}
+	return &pod, nil
+}
+
+func (oe *openshiftClient) getContainerLogs(ctx context.Context, namespace string, pod string, container string) (*bytes.Buffer, error) {
+	podLogOpts := corev1.PodLogOptions{
+		Container: container,
+	}
+	req := oe.K8sInterface.CoreV1().Pods(namespace).GetLogs(pod, &podLogOpts)
```

Comment:
Are we pulling this in just to get a typed interface? It should be possible to get what you need without introducing this new field.

https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#example-Client-Get